iceoryx2-bb/memory/tests/one_chunk_allocator_tests.rs

```diff
@@ -175,22 +175,21 @@ fn one_chunk_allocator_shrink_fails_when_alignment_increases() {
 }
 
 #[test]
+#[should_panic]
 fn one_chunk_allocator_shrink_non_allocated_chunk_fails() {
     const CHUNK_SIZE: usize = 128;
     const CHUNK_ALIGNMENT: usize = 1;
     let mut test = TestFixture::new();
     let sut = test.create_one_chunk_allocator();
 
-    assert_that!(
-        unsafe {
-            sut.shrink(
-                NonNull::new(1234 as *mut u8).unwrap(),
-                Layout::from_size_align_unchecked(CHUNK_SIZE, CHUNK_ALIGNMENT),
-                Layout::from_size_align_unchecked(CHUNK_SIZE / 2, CHUNK_ALIGNMENT),
-            )
-        },
-        is_err
-    );
+    unsafe {
+        sut.shrink(
+            NonNull::new(1234 as *mut u8).unwrap(),
+            Layout::from_size_align_unchecked(CHUNK_SIZE, CHUNK_ALIGNMENT),
+            Layout::from_size_align_unchecked(CHUNK_SIZE / 2, CHUNK_ALIGNMENT),
+        )
+        .unwrap();
```

Comment:
It is unclear whether the panic shall come from `shrink` or `unwrap`. You could use `let _ = sut.shrink(...` to clarify the intention better.

I'm positive surprised that this also works for release builds although a `debug_assert` is used.

Reply:
I added `RUSTFLAGS="-C debug-assertions"` to our release builds to be able to test those things.

Reply:
Hmm, doesn't this counteract the optimizations you did with debug assert?

Reply:
Okay, it is only in the CI but with that one has to keep in mind to also use it for local builds. What do you think about using `#[cfg(debug_assertions)]` on the related tests?

Reply:
Yes, but it is only for the CI, see `.cirrus.yml`.

Reply:
That's what I mean. When I run these tests locally in release mode they will fail and I'm pretty sure I will have forgotten about the flag in a few weeks.